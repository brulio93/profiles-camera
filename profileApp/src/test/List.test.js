import React from 'react';
import Enzyme from 'enzyme';
import MyList from '../components/list/List';
import {mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';

Enzyme.configure({adapter: new Adapter()});
describe('ProfilesList', function() {
    beforeEach(function(){
        this.store = {
            filteredProfiles: [
                {name: 'Test1', phone: '3332', address: 'Test address', photo_uri: 'https://octicons.github.com/img/og/mark-github.png'},
                {name: 'Test2', phone: '3332', address: 'Test address', photo_uri: 'https://octicons.github.com/img/og/mark-github.png'},
                {name: 'Test3', phone: '3332', address: 'Test address', photo_uri: 'https://octicons.github.com/img/og/mark-github.png'}
            ],
            filter: "test",
            addProfile: (val) => {
                this.createProfileCalled = true
                this.profileValue = val
            },
        }
    });

    it("render filtered profiles", function(){
        const wrapper = mount(<MyList store={this.store} />)
        expect(wrapper.find("Card ListItem").at(0).text()).toBe("Test1");
    });
});
