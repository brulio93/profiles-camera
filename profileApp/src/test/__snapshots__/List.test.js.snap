// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfilesList render filtered profiles 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "borderColor": "#e1e8ee",
            "borderWidth": 1,
            "margin": 15,
            "marginBottom": 0,
            "padding": 15,
            "shadowColor": "rgba(0,0,0, .2)",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 1,
            "shadowRadius": 1,
          },
          undefined,
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
            undefined,
          ]
        }
      >
        <View>
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {},
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Array [
                  Object {
                    "color": "#43484d",
                    "fontSize": 17.5,
                    "fontWeight": "bold",
                    "marginBottom": 15,
                    "textAlign": "center",
                  },
                  undefined,
                  undefined,
                  undefined,
                ],
              ]
            }
          >
            List of Contacts
          </Text>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#e1e8ee",
                  "height": 1,
                },
                Array [
                  Object {
                    "marginBottom": 15,
                  },
                  undefined,
                ],
              ]
            }
          />
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
